// @storybook/builder-vite/dist/plugins/mdx-plugin.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.mdxPlugin = void 0;
const isStorybookMdx = (id) => id.endsWith('stories.mdx') || id.endsWith('story.mdx');
/**
 * Storybook uses two different loaders when dealing with MDX:
 *
 * - *stories.mdx and *story.mdx are compiled with the CSF compiler
 * - *.mdx are compiled with the MDX compiler directly
 *
 * @see https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/recipes.md#csf-stories-with-arbitrary-mdx
 */
function mdxPlugin(options) {
    const { features } = options;
    return {
        name: 'storybook-vite-mdx-plugin',
        enforce: 'pre',
        async transform(src, id) {
            if (id.match(/\.mdx?$/)) {
                // @ts-ignore
                const { compile } = (features === null || features === void 0 ? void 0 : features.previewMdx2)
                    ? await Promise.resolve().then(() => __importStar(require('@storybook/mdx2-csf')))
                    : await Promise.resolve().then(() => __importStar(require('@storybook/mdx1-csf')));
                // TODO: we don't currently support setting mdx options.  Storybook 7.0 does
                const mdxCode = String(await compile(src, { skipCsf: !isStorybookMdx(id) }));
                return {
                    code: mdxCode,
                    map: null,
                };
            }
        },
    };
}
exports.mdxPlugin = mdxPlugin;
